src/common/meta/src/ddl/create_table.rs

```diff
@@ -294,8 +316,36 @@ impl TableCreator {
                 task,
                 region_routes,
             },
+            opening_regions: vec![],
         }
     }
+
+    /// Register opening regions if doesn't exist.
+    pub fn register_opening_regions(&mut self, context: &DdlContext) -> Result<()> {
+        let region_routes = &self.data.region_routes;
+
+        let opening_regions = operating_leader_regions(region_routes);
+
+        if self.opening_regions.len() == opening_regions.len() {
+            return Ok(());
+        }
+
+        let mut opening_region_guards = Vec::with_capacity(opening_regions.len());
+
+        for (region_id, datanode_id) in opening_regions {
+            let guard = context
+                .memory_region_keeper
+                .register(datanode_id, region_id)
+                .context(error::RegionOperatingRaceSnafu {
```

Comment:
Under what conditions could this race be happened?

Reply:
In theory, this is impossible. However, it requires acquiring locks carefully. So, I keep this checking here.